### PR TITLE
support  ip and dnsname  to number, use to_number

### PR DESCRIPTION
**Pull request by hktalent**
support  ip and dnsname  to number, use to_number

### CODE REVIEW

**Review by ehsandeep — COMMENTED** [review]
Hi @hktalent,

Thanks for opening this PR, ideally it would be nice to have issue associated with PR to understand the change, if that is not the case, please describe the relevant information in the PR itself.

Could you please confirm what problem you looking to fix / support with this PR?

**Comment by hktalent** [comment]
@ehsandeep 
In the actual use of yaml, sometimes we need to convert the ip or domain to a number, and then use it in the payload, or match it in the result

**Comment by forgedhallpass** [comment]
@hktalent

I'd like to share with you some good practices:

1. One of the most obvious ways to showcase what you want to achieve, especially in case of the DSL helper functions is to extend the existing test cases. Having tests would also help with protecting the functionality from bugs resulted by refactorings etc.
2. In order to make the PR verification/validation and merging smoother, please make sure your create your PR from the same branch in which you intend it to be merged. In this case, the PR would only show the relevant changes and commits from you.
3. Please do not add comments like "added by XYZ" into the code base. Your contributions, once accepted and merged will be visible in the git history, and also under the project contributors on GitHub.

**Comment by hktalent** [comment]
@forgedhallpass ok

File: v2/pkg/operators/common/dsl/dsl.go [hunk 1]
```diff
@@ -490,6 +492,32 @@ func init() {
 			} else if govalidator.IsFloat(argStr) {
 				sint, err := strconv.ParseFloat(argStr, 64)
 				return float64(sint), err
+			} else if govalidator.IsIP(argStr) { // add by 51pwn
+				ip := net.ParseIP(argStr)
+				i := big.NewInt(0)
+				i.SetBytes(ip)
+				return i.Int64(), nil
+			} else if govalidator.IsDNSName(argStr) { // add by 51pwn
```
Comment (Mzack9999) [inline v2/pkg/operators/common/dsl/dsl.go:500]: What about adding two independent helper functions called for example `dns_to_ip` and `ip_to_number`. Could you also provide a bit more context about the use cases?

**Review by Mzack9999 — CHANGES_REQUESTED** [review]
Suggesting clearer helper functions naming

File: v2/pkg/operators/common/dsl/dsl.go (hunk 1 above)
Comment (hktalent) [reply v2/pkg/operators/common/dsl/dsl.go:500]: yes，thanks

Comment (Mzack9999) [reply v2/pkg/operators/common/dsl/dsl.go:500]: The helper functions seem to be very specific. Would it be possible to provide more context about their use cases? For which scenarios do you need to resolve a domain name to IP and convert it to a number?

Comment (hktalent) [reply v2/pkg/operators/common/dsl/dsl.go:500]: @Mzack9999
scene

1. Check whether the response contains IP_ to_ number(xx), ip_ to_ Special value for number (dns_to_ip (XX))

2. IP this data_ to_ number(xx), ip_ to_ Number (dns_to_ip (XX)) is implicitly sent to other systems for verification

3. Use the converted number to make a specific payload

**Pull request closed without merging**